Sources/Sentry/SentrySDK.m

```diff
@@ -53,6 +63,30 @@ + (void)setCrashedLastRunCalled:(BOOL)value
     crashedLastRunCalled = value;
 }
 
+/**
+ * Not public, only for internal use.
+ */
++ (NSObject *)appStartMeasurementLock
+{
+    return appStartMeasurementLock;
+}
+
+/**
+ * Not public, only for internal use.
+ */
++ (SentryAppStartMeasurement *)appStartMeasurement
+{
+    return appStartMeasurement;
+}
+
+/**
+ * Not public, only for internal use.
+ */
++ (void)setAppStartMeasurement:(SentryAppStartMeasurement *)value
+{
+    appStartMeasurement = value;
+}
+
```

Comment:
We expose a lock and a getter and setter.
The contract then is: Please caller, make sure to take the lock before getting or setting. But we don't use the lock when setting at the moment (https://github.com/getsentry/sentry-cocoa/pull/1111/files#diff-e123bb68e38d254b173a46592ea2739c36f9d042da0bf0f7e52384954903ae33R122)

The lock is used when getting so we can do a check 'is it null' no? Then take and set to null.

If I understood correctly, another (simpler) approach here is to expose only the getter and setter here. Use the lock object internally to have exclusive access to the field and change the getter to be named something like `takeAppStartMeasurement`. To clarify the behavior is: If you call me, and there's an instance, I'll take the lock, set the field to null, and return that instance so subsequent calls will return null.

Reply:
I totally agree with your point. How didn't I see this before. Unfortunately, with the current approach, the code is easy to misuse.